### PR TITLE
Single Precision Tests

### PR DESCRIPTION
**Pull request by ax3l**
Add support for single precision tests with appropriate error margins.

- [x] depends on #422
- [x] depends on https://github.com/AMReX-Codes/amrex/pull/683

### CODE REVIEW

File: Regression/WarpX-tests.ini [hunk 1]
```diff
@@ -248,6 +248,24 @@ particleTypes = electrons
 analysisRoutine = Examples/Tests/Langmuir/analysis_langmuir2d.py
 analysisOutputImage = langmuir2d_analysis.png
 
+[Langmuir_2d_single_precision]
+buildDir = .
+inputFile = Examples/Tests/Langmuir/inputs_3d_rt
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]: todo: clarify from other examples

File: Examples/Tests/Langmuir/analysis_langmuir2d.py [hunk 2]
```diff
@@ -50,7 +56,7 @@
            %np.max( np.abs( ( Ex[:32,:,0] - E_predicted ) / E_predicted ) ) )
 assert np.allclose( Ex[:32,:,0], E_predicted, rtol=0.1 )
 print( "absolute error: np.max( np.abs( Ex[32:,:,0] ) ) = %s" %np.max( np.abs( Ex[32:,:,0] ) ) )
-assert np.allclose( Ex[32:,:,0], 0, atol=1.e-4 )
+assert np.allclose( Ex[32:,:,0], 0, atol=np.abs(E_predicted)*dfloat )
```
Comment (ax3l) [inline Examples/Tests/Langmuir/analysis_langmuir2d.py]: Maybe:
```suggestion
assert np.allclose( Ex[32:,:,0], 0, atol=dfloat )
```
?

File: Regression/WarpX-tests.ini [hunk 3]
```diff
@@ -338,6 +356,24 @@ particleTypes = electrons positrons
 analysisRoutine = Examples/Tests/Langmuir/analysis_langmuir_multi.py
 analysisOutputImage = langmuir_multi_analysis.png
 
+[Langmuir_multi_single]
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]:
```suggestion
[Langmuir_multi_single_precision]
```

File: Regression/WarpX-tests.ini [hunk 4]
```diff
@@ -338,6 +356,24 @@ particleTypes = electrons positrons
...
+numprocs = 4
+useOMP = 1
+numthreads = 2
+compileTest = 0
+doVis = 0
+compareParticles = 1
+runtime_params = warpx.do_dynamic_scheduling=0
+particleTypes = electrons positrons
+analysisRoutine = Examples/Tests/Langmuir/analysis_langmuir_multi.py
+analysisOutputImage = langmuir_multi_analysis.png
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]: Maybe unique name?
```suggestion
analysisOutputImage = langmuir_multi_analysis_single_precision.png
```

File: Regression/WarpX-tests.ini [hunk 5]
```diff
@@ -248,6 +248,24 @@ particleTypes = electrons
...
+useMPI = 1
+numprocs = 2
+useOMP = 1
+numthreads = 2
+compileTest = 0
+doVis = 0
+compareParticles = 1
+particleTypes = electrons
+analysisRoutine = Examples/Tests/Langmuir/analysis_langmuir2d.py
+analysisOutputImage = langmuir2d_analysis.png
```
Comment (ax3l) [inline Regression/WarpX-tests.ini]: Maybe unqiue name?
```suggestion
analysisOutputImage = langmuir2d_analysis_single_precision.png
```

**Comment by MaxThevenet** [comment]
Ah, also, tests don't pass, is this just a matter of merging dev in here?

**Comment by ax3l** [comment]
Sounds good, will keep in mind.
Since you mention a lossy compression transformation to save storage on disk: we will soon expose lossless and even lossy compression methods in the API again. Lossless compression for PIC data saves a factor 2-5x usually, and lossy can be even several orders of magnitude.

**Comment by ax3l** [comment]
@RemiLehe @MaxThevenet ready for review :)

**Pull request merged**